### PR TITLE
Maintenance/updates

### PR DESCRIPTION
**Pull request by kuhe**
Who still uses this? :p

PR:
- proposed minor update
- update dependency versions
- ghooks -> husky
- tape -> mocha
- jscs/jshint -> eslint
- +esdoc
- use npm 5+ and node 9.2 (enabling `npx` for local node_modules bins)

- fix bindComponent method to use appendChild on any non string template output i.e. HTML element nodes. (IE11 doesn't implement the class `HTMLElement` like other browsers)

### CODE REVIEW

File: package.json [hunk 1]
```diff
@@ -45,62 +52,57 @@
...
     ]
   },
-  "config": {
-    "ghooks": {
-      "pre-commit": "cat src/scripts/**/*.js | jscs --preset=airbnb; npm run lint; npm test"
-    }
-  },
   "scripts": {
-    "serve": "http-server -c-1 -p 9966",
+    "serve": "npx http-server -c-1 -p 9966",
```
Comment (gbabula) [inline package.json]: Do we need a `npx` dev dependency?

Comment (kuhe) [reply package.json]: npx is an automatic part of npm 5+ in all cases where I've used npm5+, including docker.

Comment (adampirani) [reply package.json]: My understanding is that it comes with npm v5 (https://medium.com/@maybekatz/introducing-npx-an-npm-package-runner-55f7d4bd282b)

Comment (gbabula) [reply package.json]: Ok awesome 👍 LGTM in that case

**Pull request merged**